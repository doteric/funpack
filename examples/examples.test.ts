import { readFileSync } from 'fs';
import { join } from 'path';

import funpack from '../src';

const directories = [
  {
    dir: 'basic-multi',
    expectedOutput: [
      { name: 'functionOne', zip: { approxSize: 854 } },
      { name: 'functionTwo', zip: { approxSize: 856 } },
    ],
  },
  {
    dir: 'basic-single',
    expectedOutput: [{ name: 'main', zip: { approxSize: 851 } }],
  },
  {
    dir: 'mjs-extension',
    expectedOutput: [{ name: 'example', zip: { approxSize: 701 } }],
  },
  {
    dir: 'module-splitting',
    expectedOutput: [
      { name: 'exampleTwo', zip: { approxSize: 432 } },
      { name: 'exampleOne', zip: { approxSize: 21597 } },
    ],
  },
  {
    dir: 'circular-dependency',
    expectedOutput: [{ name: 'main', zip: { approxSize: 899 } }],
  },
];

describe('examples', () => {
  const rootDir = process.cwd();
  afterEach(() => {
    process.chdir(rootDir);
  });
  for (const dirSettings of directories) {
    const { dir, expectedOutput } = dirSettings;
    it(`"${dir}" packages correctly`, async () => {
      console.log = jest.fn();
      process.chdir(`./examples/${dir}`);

      await funpack();

      // TODO: Improve test case to check for files?
      expect(console.log).toHaveBeenCalledTimes(expectedOutput.length);
      const logMock = console.log as jest.Mock;

      for (const expected of expectedOutput) {
        const packageJson = JSON.parse(
          readFileSync(join('dist', expected.name, 'package.json')).toString()
        );
        expect(packageJson).toMatchSnapshot();

        const foundCall = logMock.mock.calls.find((params) =>
          params[0].includes(expected.name)
        );

        expect(foundCall[0]).toBe(join('dist', `${expected.name}.zip`));
        expect(foundCall[2]).toBeCloseTo(expected.zip.approxSize, -2);
      }
    });
  }
});
