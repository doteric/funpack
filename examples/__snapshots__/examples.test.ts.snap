// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples "basic-multi" packages correctly 1`] = `
Object {
  "dependencies": Object {},
  "main": "index.js",
  "type": "module",
}
`;

exports[`examples "basic-multi" packages correctly 2`] = `
Object {
  "dependencies": Object {},
  "main": "index.js",
  "type": "module",
}
`;

exports[`examples "basic-single" packages correctly 1`] = `
Object {
  "dependencies": Object {},
  "main": "index.js",
  "type": "module",
}
`;

exports[`examples "circular-dependency" packages correctly 1`] = `
Object {
  "dependencies": Object {},
  "main": "index.js",
  "type": "module",
}
`;

exports[`examples "mjs-extension" packages correctly 1`] = `
Object {
  "dependencies": Object {},
  "main": "index.mjs",
}
`;

exports[`examples "module-splitting" packages correctly 1`] = `
Object {
  "dependencies": Object {},
  "main": "index.js",
  "type": "module",
}
`;

exports[`examples "module-splitting" packages correctly 2`] = `
Object {
  "dependencies": Object {
    "rimraf": "3.0.2",
  },
  "main": "index.js",
  "type": "module",
}
`;
